src/charm.py

```diff
@@ -9,15 +9,21 @@
 from uuid import uuid4
 
 import yaml
+from charmed_kubeflow_chisme.exceptions import ErrorWithStatus
```

Comment:
If we're updating to use `ErrorWithStatus`, let's use it everywhere the `CheckFailedError` is used.  They're different iterations of the same thing